### PR TITLE
Neuralnet Tracker Improvements

### PR DESCRIPTION
**Pull request by DaWelter**
@sthalik , as promised.

It's a bit big so I left the individual commits instead of squashing.

There are some nice changes, I guess. :-)

Please test if possible because I changed how the face region is determined during tracking.

Been experimenting with recurrent neural nets, hence the support for those. But still I haven't found something that I like more than the released net.

### CODE REVIEW

**Comment by sthalik** [comment]
The changes do indeed look nice from reading the description and the code. I can't test it as my only visible light webcam has broken lens. We can merge it now and do a public test build while postoponing the next release, what do you think? The user complaining about MJPEG framerate should be willing to help.

From experience, the thing to watch out for is assertion failures when user's face is right next to the edge of the screen.

**Comment by DaWelter** [comment]
> The changes do indeed look nice from reading the description and the code. I can't test it as my only visible light webcam has broken lens. We can merge it now and do a public test build while postoponing the next release, what do you think? The user complaining about MJPEG framerate should be willing to help.

Agreed.

Beware I put two more commits on top. Fixes - sort of.

> From experience, the thing to watch out for is assertion failures when user's face is right next to the edge of the screen.

This has always been fine. I think there is only one place where things could go wrong and `cv::getRectSubPix` takes care of the tricky parts.

File: tracker-neuralnet/ftnoir_tracker_neuralnet.cpp [hunk 1]
```diff
@@ -356,80 +587,138 @@ std::optional<PoseEstimator::Face> PoseEstimator::run(
...
+        // In old behaviour  ROI is taken from the model outputs
+        const vec3 offset = rotate_vec(face->rotation, vec3{0.f, 0.1f*face->size, face->size*0.3f});
+        const float halfsize = face->size/float(settings_.roi_zoom);
+        face->box = cv::Rect2f(
+            face->center.x + offset[0] - halfsize,
+            face->center.y + offset[1] - halfsize,
+            halfsize*2.f,
+            halfsize*2.f
+        );
+    }
```
Comment (DaWelter) [inline tracker-neuralnet/ftnoir_tracker_neuralnet.cpp:676]: @DaWelter This is a change I'm not so sure about ... I might decide to revert it or make it an option. Depends on what works best for future networks.

Comment (DaWelter) [reply tracker-neuralnet/ftnoir_tracker_neuralnet.cpp:676]: @sthalik , from my side you can merge it.

**Pull request merged**